src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -167,36 +170,38 @@ private IEnumerable<string> GetViewNames()
                 }
 
                 yield return "String";
+                yield break;
             }
-            else if (fieldType.IsInterface())
-            {
-                if (typeof(IEnumerable).IsAssignableFrom(fieldType))
-                {
-                    yield return "Collection";
-                }
-
-                yield return "Object";
-            }
-            else
+            else if (!fieldType.IsInterface())
             {
-                var isEnumerable = typeof(IEnumerable).IsAssignableFrom(fieldType);
-
+                var type = fieldType;
                 while (true)
                 {
-                    fieldType = fieldType.BaseType();
-                    if (fieldType == null)
+                    type = type.BaseType();
+                    if (type == null || type == typeof(Object))
                     {
                         break;
                     }
 
-                    if (isEnumerable && fieldType == typeof(Object))
-                    {
-                        yield return "Collection";
-                    }
+                    yield return type.Name;
+                }
+            }
 
-                    yield return fieldType.Name;
+            if (typeof(IEnumerable).IsAssignableFrom(fieldType))
+            {
+                if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
+                {
+                    yield return nameof(IFormFileCollection);
```

Comment:
odd for this to return `nameof(IFormFileCollection)` and not `typeof(IEnumerable<IFormFile>).Name`.  let's not introduce an inconsistency that needs to be explained.